notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix} ) ^2
+\end{aligned}
+$$
+
+In our data set, n = 3, so let's expand out this equation by breaking up the summation.
+
+$$
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&=
+(y_1 - \begin{bmatrix} 1 & x_1 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2  \\
+&+
+(y_2 - \begin{bmatrix} 1 & x_2 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2 \\
+&+
+(y_3 - \begin{bmatrix} 1 & x_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2 \\
+\end{aligned}
+$$
+
+Notice that our loss function is a sum of squares. Now, notice that the *L2*-norm for a vector has a similar form: 
+
+$$\Vert \vec{v} \Vert = \sqrt{v_1^2 + v_2^2 + \dots + v_n^2}$$ 
+
+So if we can express each $y_i - \begin{bmatrix} 1 & x_i \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}$ as a vector, we can express our loss as the *L2*-norm, squared:
+
+$$ 
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&= \left \Vert  \qquad   
+\begin{bmatrix} y_1 \\ y_2 \\ y_3  \end{bmatrix} \quad - \quad 
+\begin{bmatrix} 1 & x_1 \\ 1 & x_2 \\ 1 & x_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+X 
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+\vec{\hat{y}}
+\qquad \right \Vert ^2 \\
+\end{aligned}
+$$
+
+This gives us a new perspective on what it means to minimize the least squares error.
+
+
+$X$ and $\vec{y}$ are fixed, but $\hat{\theta_0}$ and $\hat{\theta_1}$ can take on any value, so there are *infinite possible vectors* created by matrix multiplying $X$ and $\hat{\theta}$. To have the smallest loss, we want to choose $\hat{\theta_0}$ and $\hat{\theta_1}$ such that the resulting vector is as close to $\vec{y}$ as possible. This is the same underlying logic as before: find the smallest squared distance; but between two vectors $\vec{y}$ and $\vec{\hat{y}}$, rather than considering the indefinite number of data points. Lastly, notice that the matrix multiplication is a linear combination of the columns of $X$: each $\hat{\theta_i}$ only ever multiplies with one column of $X$--this perspective shows us that our $\hat{y}$ is a linear combination of the features of our data. Let's get some geometric intuition on these observations.
+
+
+
+## Geometrical Intuition
+
+Multiplying $X$ and $\hat{\theta}$ represents adding scalar multiples of the columns of $X$: $\vec{x}$ and $\vec{1}$, resulting in a linear combination. Here is an intuition:
+
+Let's take two generic vectors, $\vec{x}$ and $\vec{1}$, colored black. We use red to represent the new vector we are adding, and blue for the older linear combinations.
```

Comment:
We use red...

This sentence should not be here because they have no idea what we're referring to at this point. I don't think it's necessary to say that they're colored black though. I think that's self-explanatory